cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -95,6 +95,52 @@ func (p *BicepProvider) EnsureConfigured(ctx context.Context) error {
 	return p.prompters.EnsureSubscriptionLocation(ctx, p.env)
 }
 
+func (p *BicepProvider) promptResourceGroupName(ctx context.Context) (string, error) {
+	// Get current resource groups
+	groups, err := p.azCli.ListResourceGroup(ctx, p.env.GetSubscriptionId(), nil)
+	if err != nil {
+		return "", fmt.Errorf("listing resource groups: %w", err)
+	}
+
+	choices := make([]string, len(groups)+1)
+	choices[0] = "Create a new resource group"
+	for idx, group := range groups {
+		choices[idx+1] = fmt.Sprintf("%d. %s", idx+1, group.Name)
+	}
+
+	choice, err := p.console.Select(ctx, input.ConsoleOptions{
+		Message: "Please pick a resource group to use:",
+		Options: choices,
+	})
+	if err != nil {
+		return "", fmt.Errorf("selecting resource group: %w", err)
+	}
+
+	if choice > 0 {
+		return groups[choice-1].Name, nil
+	}
+
+	name, err := p.console.Prompt(ctx, input.ConsoleOptions{
+		Message:      "Please enter a name for the new resource group:",
+		DefaultValue: fmt.Sprintf("rg-%s", p.env.GetEnvName()),
+	})
+	if err != nil {
+		return "", fmt.Errorf("prompting for resource group name: %w", err)
+	}
+
+	err = p.azCli.CreateOrUpdateResourceGroup(ctx, p.env.GetSubscriptionId(), name, p.env.GetLocation(),
```

Comment:
Do we need to provide pre-validation for either AZURE_RESOURCE_GROUP or tags if "select existing" is chosen?

Reply:
Wondering what you mean by validation for `AZURE_RESOURCE_GROUP`?  Like, ensuring the resource group exists?

For the tags, I will file an issue and we should discuss this.  I wonder if we want to be adding `azd-env-name` to the tags of existing resource groups.

There's also an open question about what we would do in the case where the resource group already has an `azd-env-name` tag.

We should discuss with @savannahostrowski about these adoption cases.

Reply:
Oh. I meant if the user supplied `AZURE_RESOURCE_GROUP`, or selected an existing group, should we provide some sort of pre-validation check there. A separate issue works totally fine.

Reply:
Opened #2328 to track.